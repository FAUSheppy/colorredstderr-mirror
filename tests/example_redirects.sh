#!/bin/sh

# Copyright (C) 2013-2018  Simon Ruderich
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.


echo write to stderr 1 >&2
echo write to stdout 1

(
    echo write to stdout which gets redirected to stderr
) >&2

echo write to stdout 2

(
    echo write to stderr which gets redirected to stdout >&2
) 2>&1

echo write to stdout 3

(
    (
        echo another redirect to stderr
    ) >&3
) 3>&2
