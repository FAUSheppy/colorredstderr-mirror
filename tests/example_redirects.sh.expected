>STDERR>write<STDERR<>STDERR> <STDERR<>STDERR>to<STDERR<>STDERR> <STDERR<>STDERR>stderr<STDERR<>STDERR> <STDERR<>STDERR>1<STDERR<>STDERR>
<STDERR<write to stdout 1
>STDERR>write<STDERR<>STDERR> <STDERR<>STDERR>to<STDERR<>STDERR> <STDERR<>STDERR>stdout<STDERR<>STDERR> <STDERR<>STDERR>which<STDERR<>STDERR> <STDERR<>STDERR>gets<STDERR<>STDERR> <STDERR<>STDERR>redirected<STDERR<>STDERR> <STDERR<>STDERR>to<STDERR<>STDERR> <STDERR<>STDERR>stderr<STDERR<>STDERR>
<STDERR<write to stdout 2
write to stderr which gets redirected to stdout
write to stdout 3
>STDERR>another<STDERR<>STDERR> <STDERR<>STDERR>redirect<STDERR<>STDERR> <STDERR<>STDERR>to<STDERR<>STDERR> <STDERR<>STDERR>stderr<STDERR<>STDERR>
<STDERR<